packages/shared/wallet/src/hooks/use-intercom-update.tsx

```diff
@@ -1,16 +1,63 @@
 import { useNotionalContext } from '@notional-finance/notionable-hooks';
-import { SupportedNetworks } from '@notional-finance/util';
+import {
+  getFromLocalStorage,
+  setInLocalStorage,
+  SupportedNetworks,
+} from '@notional-finance/util';
 import { useIntercom } from 'react-use-intercom';
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
+
+const DeBankAPIKey = process.env['NX_DE_BANK_API_KEY'] as string | undefined;
+
+async function fetchDeBankNetWorth(walletAddress) {
+  const url = `https://pro-openapi.debank.com/v1/user/total_balance?id=${walletAddress}`;
+
+  try {
+    const response = await fetch(url, {
+      method: 'GET',
+      headers: {
+        accept: 'application/json',
+        AccessKey: DeBankAPIKey || '',
+      },
+    });
+
+    if (!response.ok) {
+      throw new Error(`Error: ${response.status}`);
+    }
+    const data = await response.json();
+
+    return data.total_usd_value;
+  } catch (error) {
+    console.error('Failed to fetch user net worth:', error);
+  }
+}
 
 export const useIntercomUpdate = () => {
   const { update } = useIntercom();
+  const userSettings = getFromLocalStorage('userSettings');
+  const [netWorth, setNetWorth] = useState<number>(0);
   const {
     globalState: { networkAccounts, wallet },
   } = useNotionalContext();
   const isReadOnlyAddress = wallet?.isReadOnlyAddress;
   const selectedAccount = wallet?.selectedAddress;
 
+  if (
+    !userSettings.debankAddress &&
+    selectedAccount &&
+    !isReadOnlyAddress &&
+    netWorth === 0
+  ) {
+    fetchDeBankNetWorth(selectedAccount).then((netWorth) => {
+      setNetWorth(Math.trunc(netWorth));
+      setInLocalStorage('userSettings', {
+        ...userSettings,
+        debankAddress: selectedAccount,
```

Comment:
Why do we need to set either of these in local storage? The net worth will change over time.

Reply:
The debankAddress is in local storage to prevent excess api calls because we have to pay for them. The debankNetworth is in local storage so that if someone disconnects their wallet and connects again and the connection is recorded in intercom the debankNetworth is still sent and not zero